api.go

```diff
@@ -138,6 +138,11 @@ type Raft struct {
 	// the log/snapshot.
 	configurations configurations
 
+	// Holds a copy of the latest configuration which can be read
+	// independently from main loop.
+	latestConfiguration     Configuration
+	latestConfigurationLock sync.RWMutex
```

Comment:
I think this would probably work well for most use cases but there is a risk that very heavy reads of `GetConfiguration` can starve the main loop on a config change.

For example in Consul if there were some script calling `agent info` on a server that accidentally got into a hot loop and called it constantly, it could bring the whole of raft grinding to a halt. While that's hopefully not super likely it feels like a pretty bad outcome.

I think in this case `atomic.Value` might be a better way to syncronize Load and Store of this copy. Then readers can read without blocking using `latestConfiguration.Load()` and high contention might slow the leaderloop a little due to cross-processor core contention on the cache line but it can't starve it indefinitely.

What do you think?

Reply:
Thanks for the suggestion, sounds like a good solution. I forgot about the atomic package, but I too was wondering about contention. I am changing it to atomic!

Reply:
I agree that it would be better to use atomic and I will change it.